python-modules/cis_change_service/tests/test_api.py

```diff
@@ -0,0 +1,266 @@
+            },
+            follow_redirects=True
+        )
+
+        assert result.status_code == 401
+
+    @mock.patch('cis_change_service.idp.get_jwks')
+    def test_stream_bypass_publishing_mode_it_should_succeed(self, fake_jwks):
+        os.environ['CIS_STREAM_BYPASS'] = 'true'
+        f = FakeBearer()
+        fake_jwks.return_value = json_form_of_pk
+        token = f.generate_bearer_without_scope()
+        api.app.testing = True
+        self.app = api.app.test_client()
+        fh = open('tests/fixture/valid-profile.json')
+        user_profile = json.loads(fh.read())
+        fh.close()
+        user_profile = user_profile
+        result = self.app.post(
+            '/change',
+            headers={
+                'Authorization': 'Bearer ' + token
+            },
+            data=json.dumps(user_profile),
+            content_type='application/json',
+            follow_redirects=True
+        )
+
+        response = json.loads(result.get_data())
+        assert response.get('sequence_number') is not None
+        assert response.get('status_code') == 200
+        assert result.status_code == 200
+
+        status_endpoint_result = self.app.get(
+            '/status',
+            headers={
+                'Authorization': 'Bearer ' + token
+            },
+            query_string={'sequence_number': response.get('sequence_number')},
+            follow_redirects=True
+        )
+
+        assert json.loads(status_endpoint_result.get_data().decode('utf-8')).get('identity_vault') is True
+
+    @mock.patch('cis_change_service.idp.get_jwks')
+    def test_change_endpoint_fails_with_invalid_token_and_jwt_validation_false(self, fake_jwks):
+        os.environ['CIS_JWT_VALIDATION'] = 'false'
+        f = FakeBearer()
+        bad_claims = {
+            'iss': 'https://auth-dev.mozilla.auth0.com/',
+            'sub': 'mc1l0G4sJI2eQfdWxqgVNcRAD9EAgHib@clients',
+            'aud': 'https://hacks',
+            'iat': (datetime.utcnow() - timedelta(seconds=3100)).strftime('%s'),
+            'exp': (datetime.utcnow() - timedelta(seconds=3100)).strftime('%s'),
+            'scope': 'read:allthething',
+            'gty': 'client-credentials'
+        }
+        fh = open('tests/fixture/valid-profile.json')
+        user_profile = json.loads(fh.read())
+        fh.close()
+        user_profile = user_profile
+        fake_jwks.return_value = json_form_of_pk
+        token = f.generate_bearer_with_scope('read:profile', bad_claims)
+        api.app.testing = True
+        self.app = api.app.test_client()
+        result = self.app.get(
+            '/change',
+            headers={
+                'Authorization': 'Bearer ' + token
+            },
+            data=json.dumps(user_profile),
+            content_type='application/json',
+            follow_redirects=True
+        )
+
+        assert result.status_code == 200
+
+    def teardown_class(self):
+        subprocess.Popen(['killall', 'node'])
```

Comment:
i would rather not do that since it kills all node processes.
instead we should trap and kill childs, or record and kill pids

theres fanciers ways (including trapping a signal), but heres a straigthforward way (that does not work on Windows, but works everywhere else):
```
self.dynaliteprocess =subprocess.Popen("dynalite, etc", preexec_fn=os.setsid)
....
def __delete__(self):
    os.killpg(os.getpgid(self.dynaliteprocess.pid), 15)

```

Reply:
same for other files

Reply:
This is great.

Reply:
i forgot to note that this part is the reason i blocked on the review, the rest are style, comments, or just "hey did you think of that"